### ISSUE TITLE
Bug: Build Config - Docker file path gets preferred over custom docker file 

### ISSUE THREAD

**Issue by yuvii** [devtron-labs/devtron#6049]
### 📜 Description

If there's a dockerfile path configured in the build stage, if you switch to "Create Dockerfile" option the deployment would still try to build from the file path provided.

### 👟 Reproduction steps

Create an app, in the build configuration enter a path for a Dockerfile, then switch to "Create Dockerfile". The deployment would then look for the file specified in the path, ignoring the created dockerfile.

![image](https://github.com/user-attachments/assets/2ec30572-89de-4bf9-b2de-92791725b1d8)
![image](https://github.com/user-attachments/assets/d6d80f57-a1a0-4656-9e90-b8bfc7da46e7)

### 👍 Expected behavior

The deployment should use the selected dockerfile, regardless if there are other values available

### 👎 Actual Behavior

Devtron picks the docker image from the path if a value is provided, ignoring what is actively selected.

### ☸ Kubernetes version

EKS 1.30

### Cloud provider

<details>
AWS
</details>

### 🌍 Browser

Opera

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find any similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

**Comment by abhibhaw** [comment]
@yuvii are you experiencing the same as @deepak-devtron or something different?

**Comment by yuvii** [comment]
@abhibhaw no. He didn't understand my issue. It's not a complaint about user experience, this is a genuine bug. I inputted a path to "I have a Dockerfile" option, then I **_**switched**_** over to "Create Dockerfile". When I tried to run the pipeline, the build **_still tried to use the path_** from the other option. It was only when I deleted the value from the path that it would use the created Dockerfile.

**Comment by abhibhaw** [comment]
Hey @yuvii we tried replicating this on latest devtron version but we didn't find the issue. Can you please confirm your devtron version?

**Comment by abhibhaw** [comment]
Also @yuvii if u can confirm in your case option B's dockerfile is being used for final build or option A

**Comment by yuvii** [comment]
> Also @yuvii if u can confirm in your case option B's dockerfile is being used for final build or option A

I've since updated the version, I haven't tried recreating this issue,but I can try and check and see if it's still relevant

**Comment by satyampsoni** [comment]
Hi @yuvii, have you had a chance to replicate this issue? Does it still persist, or do you need any further assistance from us?

**Comment by yuvii** [comment]
@satyampsoni yes I've been able to recreate it. Just follow this steps:

1. choose "I have a Dockerfile". input a fake path. Save.
2. switch to "Create a Dockerfile" and save.  
3. Run build. 

Even though "Create" option is selected, when running the build it looks for the Dockerfile path set in the first option that is currently not active.

**Comment by abhibhaw** [comment]
Hey @yuvii , I tried replicating this:

<img width="778" alt="Image" src="https://github.com/user-attachments/assets/0a8fbda0-820d-45d7-8585-60b477464b2f" />

<img width="778" alt="Image" src="https://github.com/user-attachments/assets/6663c31a-00b8-4d14-9c51-8892ea2144ca" />

<img width="746" alt="Image" src="https://github.com/user-attachments/assets/da3c90bd-a0a8-47c1-8e36-104e15e8038d" />

Seems like it's working for me. Am I missing something? If you're still facing this issue, please help me with some more details, so that I can get this debugged and fixed. (A video replicating the issue may be helpful)

**Comment by yuvii** [comment]
@abhibhaw  I've made a video showcasing the issue:
https://www.loom.com/share/c1d89c3ea3ce404988f405348d44680a?sid=48168098-3eb8-4355-92d4-dc507458b00d

**Comment by abhibhaw** [comment]
Thanks @yuvii got your concerns.
